eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/NatsController.kt

```diff
@@ -0,0 +1,55 @@
+package com.arsiu.eduhub.controller.nats
+
+import com.google.protobuf.GeneratedMessageV3
+import com.google.protobuf.Parser
+import io.nats.client.Connection
+
+/**
+ * A generic interface for NATS controllers that handle messages using Protocol Buffers.
+ *
+ * @param ReqT The type of the request message.
+ * @param RepT The type of the reply message.
+ */
+interface NatsController<ReqT :  GeneratedMessageV3, RepT : GeneratedMessageV3> {
+
+    /**
+     * The NATS connection used for communication.
+     */
+    val connection: Connection
+
+    /**
+     * The NATS subject to publish and subscribe to.
+     */
+    val subject: String
+
+    /**
+     * The Protocol Buffers parser to parse bytes to proto and visa verse.
+     */
+    val parser: Parser<ReqT>
+
+    /**
+     * Handles an incoming request message and returns a reply message.
+     *
+     * @param request The incoming request message.
+     * @return The reply message.
+     */
+    fun handler(request: ReqT): RepT
+
+    /**
+     * Generates a success response message based on the provided object.
+     *
+     * @param obj The object to include in the response message.
+     * @return The success response message.
+     */
+    fun getSuccessResponse(obj: Any): RepT
+
+    /**
+     * Generates a failure response message with the specified exception and message.
+     *
+     * @param exception The exception type or identifier.
+     * @param message The error message.
+     * @return The failure response message.
+     */
+    fun getFailureResponse(exception: String, message: String): RepT
```

Comment:
Why do we need this function in the interface?

Reply:
While writing this, I considered standardizing the way the controller provides responses for success and failure. Also, I thought that this resembles to some kind of Subscriber(Reactive Stream Specifications).
P.S. I know that it's not a Subscriber but it's more like a callable handle in the dispatcher, but to me, it looks similar.
<img width="713" alt="image" src="https://github.com/ArSiu/eduHub/assets/33807305/59304359-9faf-47e1-9628-4a54b0359317">

but if there is an issue pls correct me, because for me it looks not bad

Reply:
You've created the contract. But this contract is not used outside of a particular class. It doesn't make any sense.

Let each implementer manage response by itself

Reply:
Understood